python/pyspark/mllib/linalg/__init__.py

```diff
@@ -841,6 +930,41 @@ def parse(s):
     def zeros(size):
         return DenseVector(np.zeros(size))
 
+    @staticmethod
+    def equals(v1_indices, v1_values, v2_indices, v2_values):
+        """
+        Check equality between sparse/dense vectors
+
+        >>> indices = [1, 2, 4]
+        >>> values = [1., 3., 2.]
+        >>> Vectors.equals(indices, values, list(range(5)), [0., 1., 3., 0., 2.])
+        True
+        >>> Vectors.equals(indices, values, list(range(5)), [0., 3., 1., 0., 2.])
+        False
+        >>> Vectors.equals(indices, values, list(range(5)), [0., 3., 0., 2.])
+        False
+        >>> Vectors.equals(indices, values, list(range(5)), [0., 1., 3., 2., 2.])
+        False
+        """
+        v1_size = len(v1_values)
+        v2_size = len(v2_values)
+        k1 = 0
+        k2 = 0
+        all_equal = True
+        while all_equal:
+            while k1 < v1_size and v1_values[k1] == 0:
+                k1 += 1
+            while k2 < v2_size and v2_values[k2] == 0:
+                k2 += 1
+
+            if k1 >= v1_size or k2 >= v2_size:
+                return k1 >= v1_size and k2 >= v2_size
+
+            all_equal = v1_indices[k1] == v2_indices[k2] and v1_values[k1] == v2_values[k2]
```

Comment:
Actually, this is OK since https://github.com/apache/spark/blob/master/python/pyspark/mllib/linalg/__init__.py#L489 checks for that. Could you please document this assumption though?